src/providers/MakerHooksProvider.js

```diff
@@ -1,8 +1,21 @@
 import React, { createContext } from 'react';
+import { instantiateMaker } from '../maker';
+import store from 'store';
 
 export const MakerObjectContext = createContext();
 
-function MakerHooksProvider({ children, maker }) {
+function MakerHooksProvider({ children, rpcUrl, addresses }) {
+  const [maker, setMaker] = React.useState(null);
+
+  React.useEffect(() => {
+    if (!rpcUrl || !addresses) return;
+    instantiateMaker({ rpcUrl, addresses }).then(maker => {
+      store.dispatch({ type: 'addresses/set', payload: { addresses } });
```

Comment:
a lot of the low-level logic in the PR was just relocated, cool to talk about ways to improve, but we probably can separate the concerns for now